### PR TITLE
refactor(RHTAPREL-642): support data JSON file in fbc-release

### PR DESCRIPTION
**Pull request by theflockers**
Signed-off-by: Leandro Mendes <lmendes@redhat.com>

### CODE REVIEW

File: pipelines/fbc-release/fbc-release.yaml [hunk 1]
```diff
@@ -4,7 +4,7 @@ kind: Pipeline
 metadata:
   name: fbc-release
   labels:
-    app.kubernetes.io/version: "0.26.0"
+    app.kubernetes.io/version: "0.27.0"
```
Comment (johnbieren) [inline pipelines/fbc-release/fbc-release.yaml:7]: I've been going to `1.0.0` for these changes. Not 100% sure when this is appropriate, but if this change isn't one that bumps the major version, I don't know what is

**Review by johnbieren — COMMENTED** [review]
LGTM, but I think we should wait for Happy's task update before merging and it would be good if another approves as well

File: pipelines/fbc-release/fbc-release.yaml [hunk 2]
```diff
@@ -32,46 +32,9 @@ spec:
...
-      description: Secret that contains IIB's service configuration
-    - name: iibOverwriteFromIndexCredential
-      default: "iib-overwrite-fromimage-credentials"
-      type: string
-      description: Secret that stores IIB's overwrite_from_index_token parameter value
-    - name: fbcPublishingCredentials
-      type: string
-      default: "fbc-publishing-credentials"
-      description: Secret used to publish the built index image
     - name: requestUpdateTimeout
```
Comment (scoheb) [inline pipelines/fbc-release/fbc-release.yaml:35]: why are we treating this differently?

**Comment by mmalina** [comment]
@theflockers again, the commit title should be an order, so maybe: `refactor(RHTAPREL-642): support data JSON file in fbc-release`

**Comment by mmalina** [comment]
This really requires https://github.com/redhat-appstudio/release-service-catalog/pull/239/files - both of these need to be merged in sync, right? I wonder if in the future this should be done together - because without specifying versions when referencing tasks, you're always using the latest. I understand that maybe the PR would be too big. So at least we should make it clear that these two PRs need to be merged at the same time.

File: pipelines/fbc-release/fbc-release.yaml (hunk 2 above)
Comment (theflockers) [reply pipelines/fbc-release/fbc-release.yaml]: just a cascading mistake. Fixed.

**Comment by theflockers** [comment]
> This really requires https://github.com/redhat-appstudio/release-service-catalog/pull/239/files - both of these need to be merged in sync, right? I wonder if in the future this should be done together - because without specifying versions when referencing tasks, you're always using the latest. I understand that maybe the PR would be too big. So at least we should make it clear that these two PRs need to be merged at the same time.

Yes. This one depends on the one.

**Review by scoheb — DISMISSED** [review]

File: pipelines/fbc-release/README.md [hunk 3]
```diff
@@ -12,20 +12,17 @@ Tekton release pipeline to interact with FBC Pipeline
...
-| buildTimeoutSeconds             | Max seconds to wait until the build finishes                                                             | Yes       | -                                               |
 | verify_ec_task_git_url          | The git repo url of the verify-enterprise-contract task                                                  | No        | -                                               |
 | verify_ec_task_git_revision     | The git repo revision the verify-enterprise-contract task                                                | No        | -                                               |
 | verify_ec_task_git_pathInRepo   | The location of the verify-enterprise-contract task in its repo                                          | No        | -                                               |
+| postCleanUp                     | Cleans up workspace after finishing executing the pipeline                                               | Yes       | true                                            |
+
+### Changes since 0.26.0
+- changes in the pipeline and tasks definitions to support usage of data JSON file
```
[the remark this answers is not in the document]
Comment (davidmogar) [reply pipelines/fbc-release/README.md:21]: Yep, imperative mood should be used as well for the PR and the commit title. Mind to fix it?

**Comment by theflockers** [comment]
I'll move this to draft for a bit while I test the full Pipeline to stop the noise.

File: pipelines/fbc-release/README.md [hunk 4]
```diff
@@ -12,20 +12,17 @@ Tekton release pipeline to interact with FBC Pipeline
...
+- removes parameters `fromIndex`, `targetIndex`, `binaryImage`, `buildTags`, `addArches`
```
[the remark this answers is not in the document]
Comment (theflockers) [reply pipelines/fbc-release/README.md]: fixed

File: pipelines/fbc-release/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (theflockers) [reply pipelines/fbc-release/README.md]: fixed

File: pipelines/fbc-release/fbc-release.yaml [hunk 5]
```diff
@@ -159,9 +119,9 @@ spec:
         resolver: "git"
         params:
           - name: url
-            value: https://github.com/redhat-appstudio/release-service-catalog.git
+            value: https://github.com/johnbieren/release-service-catalog.git
```
Comment (scoheb) [inline pipelines/fbc-release/fbc-release.yaml:122]: reminder: needs to change before merge

File: pipelines/fbc-release/fbc-release.yaml [hunk 6]
```diff
@@ -252,25 +215,21 @@ spec:
...
       workspaces:
-        - name: input
+        - name: data
           workspace: release-workspace
       taskRef:
         resolver: "git"
         params:
           - name: url
-            value: https://github.com/redhat-appstudio/release-service-catalog.git
+            value: https://github.com/theflockers/release-service-bundles.git
```
Comment (scoheb) [inline pipelines/fbc-release/fbc-release.yaml:230]: reminder: needs to change before merge

File: pipelines/fbc-release/fbc-release.yaml [hunk 7]
```diff
@@ -388,15 +333,15 @@ spec:
...
```
Comment (scoheb) [inline pipelines/fbc-release/fbc-release.yaml:342]: reminder: needs to change before merge

File: pipelines/fbc-release/fbc-release.yaml (hunk 5 above)
Comment (theflockers) [reply pipelines/fbc-release/fbc-release.yaml]: fixed

File: pipelines/fbc-release/fbc-release.yaml (hunk 6 above)
Comment (theflockers) [reply pipelines/fbc-release/fbc-release.yaml]: fixed

File: pipelines/fbc-release/fbc-release.yaml [hunk 8]
```diff
@@ -333,15 +280,15 @@ spec:
...
```
[the remark this answers is not in the document]
Comment (theflockers) [reply pipelines/fbc-release/fbc-release.yaml]: fixed

File: pipelines/fbc-release/fbc-release.yaml (hunk 7 above)
Comment (theflockers) [reply pipelines/fbc-release/fbc-release.yaml]: fixed

**Comment by theflockers** [comment]
Version tested and working with the new versions of`collect-data`, `update-ocp-tag` and `create-internal-request`.

**Review by happybhati — APPROVED** [review]

File: pipelines/fbc-release/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (mmalina) [reply pipelines/fbc-release/README.md]: PR title is still not correct.

**Review by johnbieren — APPROVED** [review]

File: pipelines/fbc-release/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (mmalina) [reply pipelines/fbc-release/README.md]: It's ok now, thanks.

**Review by mmalina — APPROVED** [review]

**Pull request merged**